src/app/shared/components/reports-link/reports-link.component.ts

```diff
@@ -25,9 +26,17 @@ export class ReportsLinkComponent {
   @Input() reports: ReadonlyArray<ReportInfo> = [];
   @Input() disableIfUnavailable = false;
 
+  private storageService = inject(StorageService);
+
   dropdownId = uniqueId("reports-link-dropdown");
 
   openReport(report: ReportInfo): void {
-    window.open(report.url, "_blank");
+    window.open(this.addTokenToUrl(report), "_blank");
+  }
+
+  addTokenToUrl(report: ReportInfo): string {
+    const url = new URL(report.url);
```

Comment:
Alternativ könntest du auch gleich den Type von `report.url` zu `URL` anpassen (statt `string`), dann macht man nicht ein `.toString()` im ReportsService, um es dann hier wieder in ein `URL` Objekt umzuwandeln und noch einmal `.toString()` zu machen... aber das ist ein Detail.